tests/v4/test_kernel.py

```diff
@@ -0,0 +1,75 @@
+import numpy as np
+import pytest
+
+from parcels import (
+    AdvectionRK4,
+    Field,
+    FieldSet,
+    ParticleSet,
+    xgcm,
+)
+from parcels._datasets.structured.generic import datasets as datasets_structured
+from parcels.xgrid import XGrid
+from tests.common_kernels import MoveEast, MoveNorth
+
+
+@pytest.fixture
+def fieldset() -> FieldSet:
+    ds = datasets_structured["ds_2d_left"]
+    grid = XGrid(xgcm.Grid(ds))
+    U = Field("U", ds["U (A grid)"], grid, mesh_type="flat")
+    V = Field("V", ds["V (A grid)"], grid, mesh_type="flat")
+    return FieldSet([U, V])
+
+
+def test_multi_kernel_reuse_varnames(fieldset):
+    pset = ParticleSet(fieldset, lon=[0.5], lat=[0.5])
+
+    # Testing for merging of two Kernels with the same variable declared
+    def MoveEast1(particle, fieldset, time):  # pragma: no cover
+        add_lon = 0.2
+        particle_dlon += add_lon  # noqa
+
+    def MoveEast2(particle, fieldset, time):  # pragma: no cover
+        particle_dlon += add_lon  # noqa
+
+    pset.execute([MoveEast1, MoveEast2], runtime=np.timedelta64(2, "s"))
```

Comment:
is dt implicitly 1 second?

Reply:
Yep, default is `dt=np.timedelta64(1, "s")`. That is actually (also) tested here